### PR TITLE
Removed the check for the kafka topic during healthcheck

### PR DESCRIPTION
**Pull request by MihaiSorinMoisa**
Updated circle.yml
Updated the producer and consumer libraries
Reworked the healthcheck

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -11,20 +11,20 @@
...
+			"checksumSHA1": "mirNgwWRdRte0ku/g4u4fNbzQP0=",
 			"path": "github.com/Financial-Times/message-queue-gonsumer/consumer",
-			"revision": "4e173298922ba4c10721b1b5b3e0d35be7f162e9",
-			"revisionTime": "2017-03-20T10:34:32Z",
-			"version": "0.3.0",
-			"versionExact": "0.3.0"
+			"revision": "0a481ff94df47f766a74867475247eaf1ed40107",
+			"revisionTime": "2017-05-26T12:08:30Z",
+			"version": "0.4.0-xp-remove-kafka-topic-check-rc.1",
+			"versionExact": "0.4.0-xp-remove-kafka-topic-check-rc.1"
```
Comment (davidbalazs93) [inline vendor/vendor.json]: as discussed, use the new stable version for consumer and producer after they are tested

File: healthchecks.go [hunk 2]
```diff
@@ -1,133 +1,84 @@
...
+	return &Healthcheck{
+		consumerInstance: consumerInstance,
+		producerInstance: producerInstance,
+		producerType:     producerType,
+	}
+}
+
+func (hc Healthcheck) Health() func(w http.ResponseWriter, r *http.Request) {
+	if hc.producerType == proxy {
+		return fthealth.HandlerParallel("Dependent services healthcheck", "Services: kafka-rest-proxy@ucs, kafka-rest-proxy@aws", hc.consumeHealthcheck(), hc.proxyForwarderHealthcheck())
```
Comment (tamas-molnar) [inline healthchecks.go]: Please remove references to UCS.

File: healthchecks.go [hunk 3]
```diff
@@ -1,133 +1,84 @@
...
 	}
 }
 
-func (bridge BridgeApp) proxyForwarderHealthcheck() ftHealth.Check {
-	return ftHealth.Check{
+func (hc Healthcheck) proxyForwarderHealthcheck() fthealth.Check {
+	return fthealth.Check{
 		BusinessImpact:   "Forwarding messages to kafka-proxy in coco won't work. Publishing in the containerised stack won't work.",
 		Name:             "Forward messages to kafka-proxy.",
 		PanicGuide:       "https://sites.google.com/a/ft.com/ft-technology-service-transition/home/run-book-library/kafka-bridge-run-book",
```
Comment (tamas-molnar) [inline healthchecks.go]: Can you update the panic guide links to dewey?

File: healthchecks.go [hunk 4]
```diff
@@ -1,133 +1,84 @@
...
 		Severity:         1,
 		TechnicalSummary: "Forwarding messages is broken. Check if kafka-proxy in coco is reachable.",
```
Comment (tamas-molnar) [inline healthchecks.go]: Reword: `check if destination proxy is reachable`

File: healthchecks.go [hunk 5]
```diff
@@ -1,133 +1,84 @@
...
+	return fthealth.HandlerParallel("Dependent services healthcheck", "Services: kafka-rest-proxy@ucs, cms-notifier@aws", hc.consumeHealthcheck(), hc.httpForwarderHealthcheck())
+}
+
+func (hc Healthcheck) consumeHealthcheck() fthealth.Check {
+	return fthealth.Check{
 		BusinessImpact:   "Consuming messages through kafka-proxy won't work. Publishing in the containerised stack won't work.",
 		Name:             "Consume from UCS kafka through the proxy",
 		PanicGuide:       "https://sites.google.com/a/ft.com/ft-technology-service-transition/home/run-book-library/kafka-bridge-run-book",
 		Severity:         1,
 		TechnicalSummary: "Consuming messages is broken. Check if kafka-proxy in aws is reachable.",
```
Comment (tamas-molnar) [inline healthchecks.go]: Reword: `check if source proxy is`

File: healthchecks.go (hunk 3 above)
Comment (MihaiSorinMoisa) [inline healthchecks.go]: I have added a reminder in the Jira task to update dewey.

**Pull request merged**